File: linetrace/EV3RT/linetrace/CommandTask.cpp
#include "CommandTask.h"
#include "ETTouchSensor.h"
#include "SerialConnect.h"

namespace linetrace
{

CommandTask::CommandTask(ETTouchSensor* touch)
:startFlag(false)
{
	this->touch = touch;
	this->serial = new SerialConnect();
}

void CommandTask::run()
{
	serial->checkCommand();

	if(serial->getCommand() == 's'){
		startFlag = true;
	}
}

bool CommandTask::checkStartCommand()
{
	if(this->touch->isButtonPressed() == true){
		this->startFlag = true;
	}

	return startFlag;
}
}  // namespace linetrace

File: linetrace/linetrace/CommandTask.java
package linetrace;

import java.util.TimerTask;

public class CommandTask extends TimerTask{

	SerialConnect serial = new SerialConnect();
	TouchSensor touch;

	boolean startFlag = false;

	CommandTask(TouchSensor touch){
		this.touch = touch;
	}

	@Override
	public void run() {
		// TODO 自動生成されたメソッド・スタブ

		serial.checkCommand();

		if(serial.getCommand() == 83){
			startFlag = true;
		}else if(touch.isButtonPressed() == true){
			startFlag = true;
		}else{

		}

	}

	public boolean checkStartCommand(){
		return startFlag;
	}

}
